pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -251,25 +251,39 @@ cdef class HashTable:
 {{py:
 
 # name, dtype, null_condition, float_group
-dtypes = [('Float64', 'float64', 'val != val', True),
-          ('UInt64', 'uint64', 'False', False),
-          ('Int64', 'int64', 'val == iNaT', False)]
+dtypes = [('Float64', 'float64', 'val != val', True, 'NAN'),
+          ('UInt64', 'uint64', 'False', False, 'NAN'),
+          ('Int64', 'int64', 'val == iNaT', False, 'iNaT')]
 
 def get_dispatch(dtypes):
-  for (name, dtype, null_condition, float_group) in dtypes:
+  for (name, dtype, null_condition, float_group, na_value) in dtypes:
     unique_template = """\
         cdef:
            Py_ssize_t i, n = len(values)
            int ret = 0
-           {dtype}_t val
+           {dtype}_t val, fill_value_val, ngaps_val
            khiter_t k
            bint seen_na = 0
            {name}Vector uniques = {name}Vector()
            {name}VectorData *ud
 
         ud = uniques.data
-
+        fill_value_val = fill_value
+        ngaps_val = ngaps
+        
         with nogil:
+            # If this is a sparse structure we need to append
+            # The fill value as well assuming the ngaps are greater than 0
+
+            if ngaps_val > 0:
+                k = kh_get_{dtype}(self.table, fill_value_val)
```

Comment:
This is duplicated 3 times in this class. Would be nice to de-dupe this somehow without making it super complicated with string interpolation...